### PR TITLE
Add library for writing DataKit-based CI systems

### LINKED ISSUE #24: Pipeline syntax
We need a way to describe pipeline syntax. Initially, shell scripts are fine, but we might want something nicer. @talex5 can you describe what you have in mind and what is missing?

### PR DESCRIPTION
**Pull request by talex5**
This PR adds a new framework for building Continuous Integration pipelines. You provide an expression that says how to calculate the CI status (e.g. "Passed" or "Failed") of a GitHub branch, tag or PR. The CI will:

- Evaluate the expression on each branch, tag and open PR in your project.
- Re-evaluate it whenever any input it used changes (e.g. a new commit is added to the PR).
- Provide a web-based UI to show the current status of the CI system.
- Set status indicators on the GitHub page.

The results of each step of each build are saved to DataKit, where they can be examined using the usual Git tools. For example, you can write scripts to analyse recent build failures.

Plugins can be written to extend its behaviour, e.g. to provision test VMs or to integrate with other CI systems.

This system is currently being used to test Docker-for-Mac, Docker-for-Windows, and Moby Linux.

ci

Closes #24.

### CODE REVIEW

**Comment by samoht** [comment]
```
File "tests/test.ml", line 571, characters 6-10:
- Warning 26: unused variable unit.
```

I guess this is because the latest version of alcotest now defines that value.

File: Dockerfile.ci [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM ocaml/opam:alpine-3.4_ocaml-4.03.0
+RUN sudo apk add ncurses-dev tzdata
```
Comment (avsm) [inline Dockerfile.ci]: why is this apk add needed? a missing depext?

Comment (avsm) [inline Dockerfile.ci]: The ncurses-dev issue should be fixed now.

Comment (talex5) [inline Dockerfile.ci]: Confirmed: all you need now to get libev support is: `opam depext -i lwt conf-libev`

**Pull request merged**